src/lib.rs

```diff
@@ -42,232 +48,290 @@ struct FilterConfig {
-                redirect_path = format!("{}?{}", path_parts[0], query);    
-            }
+        if self.get_header("authorization") != "" {
+            return Action::Continue
         }
-        self.set_http_path(redirect_path.to_owned());
-
-        info!("Request received");
-        if !self.is_authorized() {
-            info!("No auth header present. Checking for cookie containing id_token");
-            let token = self.get_cookie("oidcToken");
-            if token != "" {
-                info!("Cookie found, setting auth header");
-                self.set_http_request_header("Authorization", Some(&format!("Bearer {}", token)));
-                return Action::Continue
-            }
 
-            info!("No cookie found. Checking for code in request parameters");
-            let code = self.get_code();
-            if code != "" {
-                info!("Code found. Dispatching HTTP call to token endpoint");
-                let data: String = form_urlencoded::Serializer::new(String::new())
-                    .append_pair("grant_type", "authorization_code")
-                    .append_pair("code", code.as_str())
-                    .append_pair("redirect_uri", format!("http://{}{}", host, redirect_path).as_str())
-                    .append_pair("client_id", self.config.client_id.as_str())
-                    .append_pair("client_secret", self.config.client_secret.as_str())
-                    .finish();
-                info!("Sending data to token endpoint: {}", data);
-                
-                self.dispatch_http_call(
-                    self.config.auth_cluster.as_str(), vec![
-                        (":method", "POST"),
-                        (":path", self.config.token_uri.as_str()),
-                        (":authority", self.config.auth_host.as_str()),
-                        ("Content-Type", "application/x-www-form-urlencoded"),
-                    ],
-                    Some(data.as_bytes()),
-                    vec![],
-                    Duration::from_secs(5)
-                ).unwrap();
-                return Action::Pause
-            }
+        let token = self.get_cookie(self.config.cookie_name.as_str());
+        if token != "" {
+            debug!("Cookie found, setting auth header");
+            self.set_http_request_header(self.config.target_header_name.as_str(), Some(&format!("Bearer {}", token)));
+            return Action::Continue
+        }
 
-            info!("No code found. Redirecting to auth endpoint");
-            self.send_http_response(
-                302,
-                vec![("Location", self.get_redirect_uri(format!("http://{}{}", host, path).as_str()).as_str())],
-                Some(b""),
+        // Non html requests don't need redirects
+        let has_html_accept = self.get_header("accept").find("text/html").unwrap_or(MAX);
+        if has_html_accept == MAX {
+            self.send_internal_server_error(403, "Not Authorized".to_owned());
```

Comment:
I've tested it against prometheus and they did a fetch request without including credentials. This resulted in redirects from those urls and shows CORS errors.
I'd also safeguard against the case where the cookie expires and no page reload is done (in single page apps, e.g. grafana).

Those errors also weren't that transparent when using the pusher oauth proxy.
That's how it looks there:
![image](https://user-images.githubusercontent.com/431376/107514002-54b93100-6ba9-11eb-814b-d7e27808a188.png)

It should be more transparent in that case. Maybe we could also update the response text for this response and maybe even respond with json, if we have a `application/json` content type.

I think a setup using curl won't work anymore because we now verify whether the nonce matches the one in the response from the oidc provider.
But still, we could use a regex or some pattern matching to improve it a bit.

https://docs.rs/regex/1.4.3/regex/#example-find-a-date
```
use regex::Regex;
let re = Regex::new(r"^\d{4}-\d{2}-\d{2}$").unwrap();
assert!(re.is_match("2014-01-01"));
```